### PR TITLE
Multi-region Deploy Notifications

### PR DESCRIPTION
**Pull request by bfrederix**

### CODE REVIEW

**Comment by dylanjacoby** [comment]
**Thought (Non Blocking)**
I dont really have exceedingly strong opinions for this specific repo since it's very internally facing and likely not prone to much updating past a certain point but just a few thoughts:
 * `main.go` itself was a bit hard to follow and the biggest opportunity for cognitive overhead reduction might be breaking out all of the cloud storage operations into it's own separately testable interface
 * On a similar note for `main_test.go` having table-driven tests are sometimes a bit painful in the setup but make additional test cases a breeze once they're established, as opposed to a test-function-per-case approach.

Generally this is stuff that we'll find elsewhere, but figured now was a good time to at least mention a few bits. Otherwise it's lookin good!

**Review by dylanjacoby — APPROVED** [review]

**Comment by bfrederix** [comment]
> **Thought (Non Blocking)** I dont really have exceedingly strong opinions for this specific repo since it's very internally facing and likely not prone to much updating past a certain point but just a few thoughts:
> 
> * `main.go` itself was a bit hard to follow and the biggest opportunity for cognitive overhead reduction might be breaking out all of the cloud storage operations into it's own separately testable interface
> * On a similar note for `main_test.go` having table-driven tests are sometimes a bit painful in the setup but make additional test cases a breeze once they're established, as opposed to a test-function-per-case approach.
> 
> Generally this is stuff that we'll find elsewhere, but figured now was a good time to at least mention a few bits. Otherwise it's lookin good!

This makes sense, and something I considered doing. I hesitated with the cloud operations because I looked to see if they already had an interface, but then realized it would be pretty big to mock out, but I could just make my own wrapping interface instead for the bucket operations.

I think my only hesitancy is to put a bunch of work into something we might rarely touch, like you mentioned. That's why I kept the tests kind of ugly. But if you think it would be worth investing more time in, I'm happy to make those changes.

This is all stuff I agree with for our main projects, I just felt like it might not be worth investing the time in for this one-off 🤷🏻‍♂️
